waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/client/ThriftMetastoreClientManager.java

```diff
@@ -70,6 +87,7 @@ class ThriftMetastoreClientManager implements Closeable {
     this.hiveCompatibleThriftHiveMetastoreIfaceFactory = hiveCompatibleThriftHiveMetastoreIfaceFactory;
     this.connectionTimeout = connectionTimeout;
     msUri = conf.getVar(ConfVars.METASTOREURIS);
+    impersonationEnabled = conf.getBoolean(CommonBeans.IMPERSONATION_ENABLED_KEY,false);
```

Comment:
If you break down the implementation of `ThriftMetastoreClientManager` into two strategies, you can configure which one you want to use rather than using a feature toggle for impersonation. It's cleaner too.

Reply:
Okay, I'll try the strategy mode

Reply:
Thank you!